src/MultiDimFit.cc

```diff
@@ -665,42 +680,148 @@ void MultiDimFit::doGrid(RooWorkspace *w, RooAbsReal &nll)
+            *params = snap;
+            poiVals_[0] = x;
+            poiVars_[0]->setVal(x);
+
+
+            ///////////////////////////////////////////////////////////////////////////////////////////////////
+            /////////////////// Loop over rand points for each profiled POI to get best nll ///////////////////
+
+            bool ok;
+            int n_prof_params = specifiedVars_.size();
+            std::vector<float> nll_at_alt_start_pts;
+
+            // Get vector of points to try
+            float prof_start_pt_range_max = 20.0; // Is this what we want?
+            std::vector<std::vector<float>> wc_vals_vec_of_vec = {};
+
+            // Append the defualt start pt to the list of points to try
+            std::vector<float> default_start_pt_vec;
+            for (int prof_param_idx=0; prof_param_idx<n_prof_params; prof_param_idx++) {
+                default_start_pt_vec.push_back(specifiedVars_[prof_param_idx]->getVal());
+            }
+            wc_vals_vec_of_vec.push_back(default_start_pt_vec);
+
+            // Append the random points to the vecotr of points to try
+            for (int pt_idx=0; pt_idx<pointsRandProf_; pt_idx++) {
+                std::vector<float> wc_vals_vec;
+                for (int prof_param_idx=0; prof_param_idx<n_prof_params; prof_param_idx++) {
+                    // Get a random number in the range [-prof_start_pt_range_max,prof_start_pt_range_max]
+                    float rand_num = (rand()*2.0*prof_start_pt_range_max)/RAND_MAX - prof_start_pt_range_max;
+                    wc_vals_vec.push_back(rand_num);
+                }
+                wc_vals_vec_of_vec.push_back(wc_vals_vec);
+            }
+
+            // Print vector of points to try
+            if (verbose > 1) {
+                std::cout << "List of points to try:" << std::endl;
+                for (auto vals_vec: wc_vals_vec_of_vec) {
+                    std::cout << "\tThe vals at this point: " << std::endl;
+                    for (auto val: vals_vec) {
+                        std::cout << "\t\tPoint val: " << val << std::endl;
+                    }
+                }
+            }
+
+            // Loop over starting points to try for the prof WCs
+            for (unsigned int start_pt_idx=0; start_pt_idx<wc_vals_vec_of_vec.size(); start_pt_idx++){
+                *params = snap;
+                poiVals_[0] = x;
+                poiVars_[0]->setVal(x);
+
+                // Loop over prof POIs and set their values
+                if (verbose > 1) std::cout << "\n\tStart pt idx: " << start_pt_idx << std::endl;
+                for (unsigned int var_idx=0; var_idx<specifiedVars_.size(); var_idx++){
+                    if (verbose > 1) std::cout << "\t\tThe var name: " << specifiedVars_[var_idx]->GetName() << std::endl;
+                    if (strcmp(specifiedVars_[var_idx]->GetName(),"r")==0) {
```

Comment:
Should this be skipped in general?

Reply:
This block of skipping "r" has been removed. We agreed that there is no need to skip this in general.